apps/teams-test-app/src/components/DialogAPIs.tsx

```diff
@@ -46,29 +46,71 @@ const DialogAPIs = (): ReactElement => {
     ApiWithTextInput<string>({
       name: 'sendMessageToChild',
       title: 'sendMessageToChild',
-      onClick: async message => {
-        if (childWindowRef.current && childWindowRef.current !== null) {
-          const childWindow = childWindowRef.current;
-          await childWindow.postMessage(message);
-          return 'Message sent to child';
-        } else {
-          return 'childWindow doesnt exist';
-        }
+      onClick: {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        validateInput: () => {},
+        submit: {
+          withPromise: async message => {
+            if (childWindowRef.current && childWindowRef.current !== null) {
+              const childWindow = childWindowRef.current;
+              await childWindow.postMessage(message);
+              return 'Message sent to child';
+            } else {
+              return 'childWindow doesn\'t exist';
+            }
+          },
+          withCallback: (message, setResult) => {
+            if (childWindowRef.current && childWindowRef.current !== null) {
+              const childWindow = childWindowRef.current;
+              const onComplete = (status: boolean, reason?: string): void => {
+                if (reason && !status) {
+                  setResult(JSON.stringify(reason));
+                } else {
+                  setResult('Message sent to child');
+                }
+              };
+              childWindow.postMessage(message, onComplete);
+            } else {
+              setResult("childWindow doesn't exist");
+            }
```

Comment:
Can we please make the string consistent with `withPromise` here? Also `SendMessageToParent` below.

Reply:
This was more of a lint error, the output string would be the same. The difference is we needed to use \' to get the apostraphe into the first string due to using single quotations. The second one we can't pass single quotations in. I could change it to `childWindow doesnt exist` to keep it consistent though